### PR TITLE
all: update golang/x/ext and fix slice sorting fallout

### PR DESCRIPTION
**Pull request by karalabe**
The Go authors updated golang/x/ext to change the function signature of the slices sort method. It's an entire shitshow now because x/ext is not tagged, so everyone's codebase just picked a new version that some other dep depends on, causing our code to fail building.

This PR updates the dep on our code too and does all the refactorings to follow upstream...

Fixes https://github.com/ethereum/go-ethereum/issues/27897

### CODE REVIEW

File: core/rawdb/accessors_chain.go [hunk 1]
```diff
@@ -902,9 +902,9 @@ func WriteBadBlock(db ethdb.KeyValueStore, block *types.Block) {
 		Header: block.Header(),
 		Body:   block.Body(),
 	})
-	slices.SortFunc(badBlocks, func(a, b *badBlock) bool {
+	slices.SortFunc(badBlocks, func(a, b *badBlock) int {
 		// Note: sorting in descending number order.
-		return a.Header.Number.Uint64() >= b.Header.Number.Uint64()
+		return -a.Header.Number.Cmp(b.Header.Number)
```
Comment (holiman) [inline core/rawdb/accessors_chain.go:907]:
```suggestion
		return b.Header.Number.Cmp(a.Header.Number)
```
Alternative version is to just swap the arguments. Use with whichever you feel like

File: p2p/protocol.go [hunk 2]
```diff
@@ -78,9 +79,15 @@ func (cap Cap) String() string {
 }
 
 // Less defines the canonical sorting order of capabilities.
-func (cap Cap) Less(other Cap) bool {
+func (cap Cap) Less(other Cap) int {
```
Comment (holiman) [inline p2p/protocol.go:82]:
```suggestion
func (cap Cap) Cmp(other Cap) int {
```
Wouldn't that be more correct? Adds more fallout, sorry ...

Comment (karalabe) [reply p2p/protocol.go:82]: Ah yes, indeed

**Review by holiman — APPROVED** [review]
LGTM

**Pull request merged**